[EDIT] Go: Composition - Inheritance examples

Jamesllllllllll opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

<!-- Please write a summary of the change, such as which topic(s) and file(s) that you have edited or created. Please also include relevant motivation and context: -->

Added examples of inheriting methods and fields with `struct` embedding. Added `pseudo` code block, a brief `go` code snippet showing how to add it to Pizza Restaurant example, a final `go` code snippet with the full example and the output.

Closes #3400 

### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Editing an existing entry (adding examples of inheritance via composition)

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

caupolicandiaz commented on content/go/concepts/composition/composition.md:
```diff
@@ -133,6 +133,129 @@ Margherita pizza is a medium pizza with toppings of: [tomatoes mozzarella basil]
 
+## Inheritance through Composition
+
+A `struct` can 'inherit' the fields and methods of an embedded `struct` through composition rather than traditional [inheritance](https://www.codecademy.com/resources/docs/general/programming-paradigms/inheritance).
```
```suggestion
A struct can 'inherit' the fields and methods of an embedded struct through composition rather than traditional [inheritance](https://www.codecademy.com/resources/docs/general/programming-paradigms/inheritance).
```

caupolicandiaz replied:
The struct referenced here doesn't need to be inside backticks. This formatting is used to highlight something referenced in the code examples, such as a variable. Or to offset specific syntax, such as declaring a struct requires the `struct` keyword. This formatting is incorrectly used several times in the entry, please revert the other examples as well.

caupolicandiaz commented on content/go/concepts/composition/composition.md:
````diff
@@ -133,6 +133,129 @@ Margherita pizza is a medium pizza with toppings of: [tomatoes mozzarella basil]
+}
+```
+
+Consider the following addition to the Pizza Restaurant example:
````
```suggestion
Consider the following addition to the pizza example:
```

caupolicandiaz commented on content/go/concepts/composition/composition.md:
````diff
@@ -133,6 +133,129 @@ Margherita pizza is a medium pizza with toppings of: [tomatoes mozzarella basil]
+}
+```
+
+An `Address` `struct` is created, and a `printAddress()` method takes an `Address` as its receiver. The anonymous field `Address` is added to `Restaurant` which allows access to the `Address` fields and methods.
````
```suggestion
An `Address` struct is created, and a `printAddress()` method takes an `Address` as its receiver. The anonymous field `Address` is added to `Restaurant` which allows access to the `Address` fields and methods.
```

caupolicandiaz commented on content/go/concepts/composition/composition.md:
````diff
@@ -133,6 +133,129 @@ Margherita pizza is a medium pizza with toppings of: [tomatoes mozzarella basil]
+
+The following example adds the `Address` `struct` and method:
+
+```go
````
This code is getting is bloated. I think it's preferable to just show the main function and perhaps have a comment at the top that this builds on the previous code

Jamesllllllllll replied:
How about this wording for the comment?

caupolicandiaz reviewed (changes requested):
@Jamesllllllllll I've added my comments. Please update the entry, and tag me when you've finished making the changes. Thanks

Jamesllllllllll commented:
> @Jamesllllllllll I've added my comments. Please update the entry, and tag me when you've finished making the changes. Thanks

Thank you for your review @caupolicandiaz ! 

Check the comment I added - I wasn't sure if I should capitalize Struct and Method, or if that is the best wording for the comment.

caupolicandiaz reviewed (approved):
@Jamesllllllllll Just made a couple of additional direct edits. This is ready for a second review

KTom101 reviewed (approved):
@Jamesllllllllll This looks good.
